meteor/client/ui/SegmentContainer/withResolvedSegment.ts

```diff
@@ -207,6 +210,7 @@ export function withResolvedSegment<T extends IProps, IState = {}>(
 			)
 
 			const rundownOrder = RundownPlaylistCollectionUtil.getRundownOrderedIDs(props.playlist)
+			const pieces = RundownPlaylistCollectionUtil.getPiecesForParts(orderedAllPartIds)
```

Comment:
I think it would be good to put this into something like a `memoizedIsolatedAutorun`. Also, would it be possible to limit the scope of reactivity on `getPiecesForParts` like it's done on `getSegmentsAndPartsSync` @ L183 above? Ultimately, we are passing a `CalculateTimingsPiece` anyway, so no point having all of the other properties there in the first place. I think the less properties we depend on, the better?

Reply:
I've made the requested changes